spec/kitchen/lifecycle_hooks_spec.rb

```diff
@@ -27,151 +27,162 @@
+    expect_local_hook_generated_and_run(:post, { local: local_command })
+    expect_local_hook_generated_and_run(:post, local_hook)
     run_lifecycle_hooks
   end
 
   it "runs multiple local hooks" do
     config.update(pre_create: ["echo foo"], post_create: ["echo bar"])
-    lifecycle_hooks.expects(:run_command).with("echo foo", STANDARD_LOCAL_OPTIONS)
-    lifecycle_hooks.expects(:run_command).with("echo bar", STANDARD_LOCAL_OPTIONS)
+    expect_local_hook_generated_and_run(:create, { local: "echo foo" })
+    expect_local_hook_generated_and_run(:create, { local: "echo bar" })
     run_lifecycle_hooks
   end
 
   it "runs a local command with a user option" do
-    config.update(post_create: [{ local: "echo foo", user: "bar" }])
-    lifecycle_hooks.expects(:run_command).with("echo foo", {
-      cwd: "/kitchen",
-      user: "bar",
-      environment: {
-        "KITCHEN_INSTANCE_NAME" => "default-toaster-10",
-        "KITCHEN_SUITE_NAME" => "default",
-        "KITCHEN_PLATFORM_NAME" => "toaster-1.0",
-        "KITCHEN_INSTANCE_HOSTNAME" => "localhost",
-      },
-    })
+    hook = { local: "echo foo", user: "bar" }
+    config.update(post_create: [hook])
+    expect_local_hook_generated_and_run(:create, hook, non_standard_local_opts: { user: "bar" })
     run_lifecycle_hooks
   end
 
   it "runs a local command with environment options" do
-    config.update(post_create: [{ local: "echo foo", environment: { FOO: "one", BAR: "two" } }])
-    lifecycle_hooks.expects(:run_command).with("echo foo", {
-      cwd: "/kitchen",
-      environment: {
-        "FOO" => "one",
-        "BAR" => "two",
-        "KITCHEN_INSTANCE_NAME" => "default-toaster-10",
-        "KITCHEN_SUITE_NAME" => "default",
-        "KITCHEN_PLATFORM_NAME" => "toaster-1.0",
-        "KITCHEN_INSTANCE_HOSTNAME" => "localhost",
-      },
-    })
+    hook = { local: "echo foo", environment: { FOO: "one", BAR: "two" } }
+    config.update(post_create: [hook])
+    command_opts = standard_local_options.dup
+    command_opts[:environment]["FOO"] = "one"
+    command_opts[:environment]["BAR"] = "two"
+    expect_local_hook_generated_and_run(:create, hook, non_standard_local_opts: command_opts)
     run_lifecycle_hooks
   end
 
   it "runs a local command with a relative cwd option" do
-    config.update(post_create: [{ local: "echo foo", cwd: "test" }])
-    lifecycle_hooks.expects(:run_command).with("echo foo", {
-      cwd: os_safe_root_path("/kitchen/test"),
-      environment: {
-        "KITCHEN_INSTANCE_NAME" => "default-toaster-10",
-        "KITCHEN_SUITE_NAME" => "default",
-        "KITCHEN_PLATFORM_NAME" => "toaster-1.0",
-        "KITCHEN_INSTANCE_HOSTNAME" => "localhost",
-      },
-    })
+    hook = { local: "echo foo", cwd: "test" }
+    config.update(post_create: [hook])
+    expect_local_hook_generated_and_run(:create, hook, non_standard_local_opts: { cwd: "#{kitchen_root}/test" })
     run_lifecycle_hooks
   end
 
```

Comment:
As this change introduces two new options (includes and excludes) for lifecycle hooks, it would be good to add tests that cover those options.

Reply:
I agree! I am going to put some time on my calendar this month to take care of this.